android/tangram/src/com/mapzen/tangram/FontFileParser.java

```diff
@@ -54,11 +68,33 @@ private void processDocument(XmlPullParser parser) throws XmlPullParserException
             } else if ("alias".equals(parser.getName())) {
                 // Parse this alias to font to fileName
                 String aliasName = parser.getAttributeValue(null, "name");
+                String aliasName2 = Character.toUpperCase(aliasName.charAt(0)) + aliasName.substring(1);
                 String toName = parser.getAttributeValue(null, "to");
                 String weightStr = parser.getAttributeValue(null, "weight");
-                weightStr = (weightStr == null) ? "400" : weightStr;
-                String fontFilename = fontDict.get(toName + "_" + weightStr + "_normal"); // alias style is default: normal (always)
-                fontDict.put(aliasName + "_" + weightStr, fontFilename);
+                Vector<String> aliasWeights = new Vector<String>();
+                String fontFilename;
+
+                if (weightStr == null) {
+                    aliasWeights = weights;
+                } else {
+                    aliasWeights.add(weightStr);
+                }
+
+                for (int i = 0; i < aliasWeights.size(); ++i) {
+                    // Only 2 styles possible based on /etc/fonts.xml
+                    // Normal style
+                    fontFilename = fontDict.get(toName + "_" + aliasWeights.get(i) + "_normal");
+                    fontDict.put(aliasName + "_" + aliasWeights.get(i) + "_normal", fontFilename);
+                    fontDict.put(aliasName2 + "_" + aliasWeights.get(i) + "_normal", fontFilename);
+                    fontDict.put(aliasName + "_" + aliasWeights.get(i) + "_Normal", fontFilename);
+                    fontDict.put(aliasName2 + "_" + aliasWeights.get(i) + "_Normal", fontFilename);
+                    // Italic style
+                    fontFilename = fontDict.get(toName + "_" + aliasWeights.get(i) + "_italic");
+                    fontDict.put(aliasName + "_" + aliasWeights.get(i) + "_italic", fontFilename);
+                    fontDict.put(aliasName2 + "_" + aliasWeights.get(i) + "_italic", fontFilename);
+                    fontDict.put(aliasName + "_" + aliasWeights.get(i) + "_Italic", fontFilename);
+                    fontDict.put(aliasName2 + "_" + aliasWeights.get(i) + "_Italic", fontFilename);
```

Comment:
These "normal" and "Normal" should also be normalized to lowercase "normal", etc.